### PR TITLE
Add gated gaussian model with marginalized polarization

### PR DESCRIPTION
**Pull request by cdcapano**
Adds a version of the `GatedGaussian` that numerically marginalizes over polarization. Requires #3712 and #3713.

### CODE REVIEW

File: pycbc/inference/models/gated_gaussian_noise.py [hunk 1]
```diff
@@ -207,24 +316,25 @@ def get_gate_times_hmeco(self):
...
         except NoWaveformError:
             return self._nowaveform_logl()
         except FailedWaveformError as e:
             if self.ignore_failed_waveforms:
                 return self._nowaveform_logl()
             raise e
-
+        # gate input for ringdown analysis which consideres a start time
+        # and an end time
+        dgate = params['gate_window']
```
Comment (ahnitz) [inline pycbc/inference/models/gated_gaussian_noise.py:330]: This is definitely a bug, no? Youa re reading from params before it is generated on the next line. I suspect this hasn't been run in exactly this state. Perhaps some patches missing? @cdcapano

File: pycbc/inference/models/gated_gaussian_noise.py [hunk 2]
```diff
@@ -154,6 +141,128 @@ def _loglr(self):
...
+
+        Returns
+        -------
+        dict :
+            Dictionary of detector names -> FrequencySeries.
+        """
+        wfs = self.get_waveforms()
+        out = {}
+        for det, h in wfs.items():
+            invpsd = self._invpsds[det]
```
Comment (ahnitz) [inline pycbc/inference/models/gated_gaussian_noise.py:217]: CC is reporting this as unused, and it certainly looks like it is to me.

**Review by ahnitz — CHANGES_REQUESTED** [review]
See the two comments (noted from the CC report as well as potential problems)

**Comment by ahnitz** [comment]
@micamu Since @cdcapano is on leave, and after you have a chance to look at it, I would be OK to approve if someone can double check they can run this model and we manually apply the fixes to the two obvious minor errors.

File: pycbc/inference/models/gated_gaussian_noise.py (hunk 1 above)
Comment (cdcapano) [reply pycbc/inference/models/gated_gaussian_noise.py]: Yeah, that's a bug. I switched the order of the lines to fix it. We haven't actually been using this function, which is why I didn't catch it.

File: pycbc/inference/models/gated_gaussian_noise.py (hunk 2 above)
Comment (cdcapano) [reply pycbc/inference/models/gated_gaussian_noise.py]: Fixed.

**Comment by cdcapano** [comment]
@ahnitz I addressed the two issues you caught, plus a couple other that code climate was complaining about.

**Review by ahnitz — APPROVED** [review]

**Comment by ahnitz** [comment]
@cdcapano I've appoved this. Note the github action failures are due to a problem with dqsegdb, not with this PR.

**Pull request merged**